extensions/consul-config/runtime/src/main/java/io/quarkus/consul/config/runtime/ConsulConfigSourceProvider.java

```diff
@@ -47,27 +48,39 @@ private ConsulConfigSourceProvider(ConsulConfig config, ConsulConfigGateway cons
 
         List<ConfigSource> result = new ArrayList<>(keys.size());
 
+        List<Uni<?>> allUnis = new ArrayList<>();
+
         for (Map.Entry<String, ValueType> entry : keys.entrySet()) {
             String fullKey = config.prefix.isPresent() ? config.prefix.get() + "/" + entry.getKey() : entry.getKey();
-            log.debug("Attempting to look up value of key '" + fullKey + "' from Consul.");
-
-            try {
-                Optional<Response> optionalResponse = consulConfigGateway.getValue(fullKey);
-                if (optionalResponse.isPresent()) {
-                    result.add(
-                            responseConfigSourceUtil.toConfigSource(optionalResponse.get(), entry.getValue(), config.prefix));
-                } else {
-                    String message = "Key '" + fullKey + "' not found in Consul.";
-                    if (config.failOnMissingKey) {
-                        throw new RuntimeException(message);
+            allUnis.add(consulConfigGateway.getValue(fullKey).chain(new Function<Response, Uni<?>>() {
```

Comment:
why using chain here? What is async in the function? 
`map` would be simpler. For the failure, just throw the exception.

Reply:
actually, even `invoke` would work as you don't produce anything "useful" (null)

Reply:
Cool, thanks! I'll update it